bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -476,6 +481,34 @@ public State getState(Widget w) {
     }
 
     /**
+     * Converts an item state to the type the widget supports (if possible)
+     *
+     * @param w - Widget in sitemap that shows the state
+     * @param s - State of the item
+     * @return the converted state or the original if conversion was not possible
+     */
+    private State convertState(Widget w, State s) {
+        State returnState = null;
+
+        if (w instanceof SwitchImpl) {
+            if (s instanceof PercentType) {
+                returnState = ((PercentType) s).as(OnOffType.class);
+            }
+        } else if (w instanceof SliderImpl) {
+            if (s instanceof PercentType) { // catches also HSBType etc
+                returnState = ((PercentType) s).as(PercentType.class);
+            }
+        }
```

Comment:
You're right.
The real usage is more the reverse, that is using a switch or a dimmer for a color item for example and it is what ypu have covered.